doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,299 @@
+* CSRAccess - Any instruction from Zicsr.
+* EBreakDbg/EBreakExc- An ``EBREAK`` instruction that either enters debug mode (Dbg) or causes an exception (Exc).
+  Which occurs depends upon the setting of ``dcsr.ebreakm`` / ``dcsr.ebreaku`` combined with the privilege level of executed instruction.
+* ``ECALL``
+* ``MRET``
+* ``DRET``
+* ``WFI``
+* ``FENCE.I``
+* FetchError - Any instruction that saw a fetch error.
+* CompressedIllegal - Any compressed instruction with an illegal encoding.
+* UncompressedIllegal - Any uncompressed instruction with an illegal encoding.
+* CSRIllegal - Any instruction attempting a CSR access that is not allowed.
+* PrivIllegal - Illegal due to privilege level or being in/out of debug mode.
+* OtherIllegal - Any other instruction that raises an Illegal instruction exception that isn’t in the other categories.
+* None - No instruction in ID/EX stage.
+
+Stalls
+""""""
+Not all instructions can see all kinds of stalls.
+A stall category is sampled at the ID/EX stage only (as stalls in IF and WB don’t break down into categories).
+
+* Instr - A stall cause by a multi-cycle instruction.
+  This can be seen by instructions from categories:
+
+  * MUL
+  * DIV
+  * Branch
+  * Jump
+
+* LdHz - A load hazard, the instruction in ID/EX depends upon the result of a load that is awaiting a response in writeback.
+  This can be seen by instructions from categories:
+
+  * ALU
+  * Mul
+  * Div
+  * Branch
+  * Jump
+  * Load
+  * Store
+  * CSRAccess
+
+* Mem - Memory stall, the instruction in ID/EX is awaiting a prior memory request to complete before it can begin (to allow precise interrupts on a memory error response). This can be seen for all instruction categories
+
+Privilege Level
+"""""""""""""""
+
+Ibex can operate at either the M (machine) or U (user) privilege levels.
+Different aspects of the Ibex microarchitecture can be using different privilege levels at once.
+
+* Privilege level of IF stage instruction.
+* Privilege level of ID/EX stage instruction.
+* Privilege level of LSU operation (ID/EX privilege level modified by `mstatus.mprv` and `mstatus.mpp` settings).
+
+Hazards
+"""""""
+Ibex hazards all occur in the interaction between the ID and EX stage.
+
+* RAW Reg - Read after write hazard, instruction in ID/EX reads a register writeback is writing.
+  Handled with data forwarding and no stall.
+* RAW Load - Read after write hazard when reading from load destination, a special case of RAW Reg above.
+  Produces a stall (Category LdHz) and shouldn’t forward data.
+
+State Specific Behaviour
+""""""""""""""""""""""""
+Some instructions will behave differently depending upon the state of the processor (e.g. the privilege level the instruction executes at, CSR settings or whether the processor is in debug mode).
+
+* Instruction illegal in U Mode (cover execution in U and M mode).
+
+  * Mret
+  * Wfi
+  * Access to M-mode CSR.
+
+* Debug mode only instructions (cover execution in and out of debug mode).
```

Comment:
Nit: I don't think you need the "only" here? Reading quickly, it currently looks like "Debug mode only ever \<verb\>".